src/finalizer/finalizer.cpp

```diff
@@ -492,12 +492,64 @@ extern "C" HRESULT DetectSdk(LPWSTR sczSdkFeatureBandVersion, LPWSTR sczArchitec
     return hr;
 }
 
+void RemoveInstallStateFile(LPWSTR sczSdkFeatureBandVersion, LPWSTR sczPlatform)
+{
+    HRESULT hr = S_OK;
+    LPWSTR sczProgramData = NULL;
+    LPWSTR sczInstallStatePath = NULL;
+    LPWSTR sczPath = NULL;
+
+    hr = ShelGetFolder(&sczProgramData, CSIDL_COMMON_APPDATA);
+    ExitOnFailure(hr, "Failed to get shell folder.");
+
+    hr = PathConcat(sczProgramData, L"dotnet", &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat dotnet to install state path.");
+
+    hr = PathConcat(sczInstallStatePath, L"workloads", &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat workloads to install state path.");
+
+    hr = PathConcat(sczInstallStatePath, sczPlatform, &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat platform (%ls) to install state path.", sczPlatform);
+
+    hr = PathConcat(sczInstallStatePath, sczSdkFeatureBandVersion, &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat feature band (%ls) to install state path.", sczSdkFeatureBandVersion);
+
+    hr = PathConcat(sczInstallStatePath, L"installstate", &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat installstate to install state path.");
+
+    hr = PathConcat(sczInstallStatePath, L"default.json", &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat default.json to install state path.");
+
+    if (FileExistsEx(sczInstallStatePath, NULL))
+    {
+        LogStringLine(REPORT_STANDARD, "Deleting install state file: %ls", sczInstallStatePath);
+        hr = FileEnsureDelete(sczInstallStatePath);
+        ExitOnFailure(hr, "Failed to delete install state file: %ls", sczInstallStatePath);
+
+        hr = PathGetParentPath(sczInstallStatePath, &sczPath);
+        ExitOnFailure(hr, "Failed to get parent path of install state file.");
+
+        LogStringLine(REPORT_STANDARD, "Cleaning up empty workload folders.");
+        DirDeleteEmptyDirectoriesToRoot(sczPath, 0);
```

Comment:
I assume this will no-op if there are still files in the folder.  The log message would be incorrect in that case.

Reply:
Would "Removing empty workload folders" work better?

Reply:
I don't know, I'm not sure it matters much anyway.